src/Models/Hooks/useRuntimeSceneData.ts

```diff
@@ -10,33 +10,47 @@ import ViewerConfigUtility from '../Classes/ViewerConfigUtility';
 import { IADT3DViewerAdapter } from '../Constants/Interfaces';
 import {
     deepCopy,
+    getDebugLogger,
     getSceneElementStatusColor,
     parseLinkedTwinExpression
 } from '../Services/Utils';
 import {
     I3DScenesConfig,
     IBehavior,
-    IExpressionRangeVisual
+    IExpressionRangeVisual,
+    IPollingConfiguration
 } from '../Types/Generated/3DScenesConfiguration-v1.0.0';
 import useAdapter from './useAdapter';
 
+const debugLogging = false;
+const logDebugConsole = getDebugLogger('useRuntimeSceneData', debugLogging);
+
 export const useRuntimeSceneData = (
     adapter: IADT3DViewerAdapter,
     sceneId: string,
     scenesConfig: I3DScenesConfig,
-    pollingInterval: number,
     /** Optional array of layer Ids to apply SceneVisual behavior filtering */
     selectedLayerIds: string[] = null
 ) => {
     const [modelUrl, setModelUrl] = useState('');
+    const [pollingInterval, setPollingInterval] = useState(10);
+    const [lastRefreshTime, setLastRefreshTime] = useState<number>(null);
     const [sceneVisuals, setSceneVisuals] = useState<Array<SceneVisual>>([]);
     const [sceneAlerts, setSceneAlerts] = useState<Array<SceneViewBadgeGroup>>(
         []
     );
 
     const sceneData = useAdapter({
-        adapterMethod: () => adapter.getSceneData(sceneId, scenesConfig),
-        refetchDependencies: [sceneId, scenesConfig],
+        adapterMethod: (args?: { isManualRefresh: boolean }) => {
+            setLastRefreshTime(Date.now());
+            return adapter.getSceneData(
+                sceneId,
+                scenesConfig,
+                selectedLayerIds,
+                args?.isManualRefresh ?? false
```

Comment:
Instead of this null check why not make `getSceneData` default to false in case it doesn't receive this parameter or if it receives a null/undefined one as well.